crates/wash-lib/src/config.rs

```diff
@@ -24,41 +20,26 @@ pub const DEFAULT_START_PROVIDER_TIMEOUT_MS: u64 = 60_000;
 pub const DEFAULT_CTX_DIR_NAME: &str = "contexts";
 
 /// Get the path to the `.wash` configuration directory. Creates the directory if it does not exist.
-pub fn cfg_dir() -> IoResult<PathBuf> {
-    let home = dirs::home_dir().ok_or_else(|| {
-        Error::new(
-            ErrorKind::NotFound,
-            "No home directory found. Please set $HOME.",
-        )
-    })?;
+pub fn cfg_dir() -> Result<PathBuf> {
+    let home = dirs::home_dir().context("no home directory found. Please set $HOME")?;
 
     let wash = home.join(WASH_DIR);
 
     if !wash.exists() {
-        fs::create_dir_all(&wash)?;
+        fs::create_dir_all(&wash).context("failed to create .wash directory")?;
```

Comment:
How do you feel about noting the path that it failed to create? `with_context` and then using the path would help make it much clearer the file that failed to create (which might explain why it failed)